lib/ratelimiter.js

```diff
@@ -0,0 +1,132 @@
+'use strict';
+
+var THREE_HOURS = 3* 60 * 60 * 1000;
+
+/**
+ * A rate limiter to be used as an express middleware.
+ *
+ * @param {Object} options
+ * @param {Number} options.limit - Number of requests for normal rate limiter
+ * @param {Number} options.every - Interval of the normal rate limiter
+ * @param {Array} options.whitelist - IP addresses that should have whitelist rate limiting
+ * @param {Array} options.blacklist - IP addresses that should be blacklist rate limiting
+ * @param {Number} options.whitelistLimit - Number of requests for whitelisted clients
+ * @param {Number} options.whitelistEvery - Interval for whitelisted clients
+ * @param {Number} options.blacklistLimit - Number of requests for blacklisted clients
+ * @param {Number} options.blacklistEvery - Interval for blacklisted clients
+ */
+function RateLimiter(options) {
+  if (!(this instanceof RateLimiter)) {
+    return new RateLimiter(options);
+  }
+
+  if (!options){
+    options = {};
+  }
+
+  this.node = options.node;
```

Comment:
Added in: https://github.com/bitpay/insight-api/pull/467/commits/ebcc3607c281839238007cebae0c15bff9d6af76